### ISSUE TITLE
Android INTERNET access permissions not being added in AndroidManifest.xml using Cordova 9.0.0

### ISSUE THREAD

**Issue by bandhavya** [apache/cordova-android#857]
# Bug Report

## Problem
Android INTERNET access permissions not being added in AndroidManifest.xml using Cordova 9.0.0

### What is expected to happen?
in the previous cordova versions, uses-permission was added directly in the AndroidManifest.xml file without any explicity permission addition in the config.xml

### What does actually happen?
Now, the permissions are not added in the AndroidManifest.xml. We should explicitly add them from the config.xml. This does not occur with phonegap build.

## Information

Just try to add a plugin which will make use of INTERNET access. On adding this plugin with cordova version less than 8.1.0 there are no conflicts. But with cordova 9.0.0, we observe that network calls are failing as there is no internet access permission included in AndroidManifest.xml file.

### Version information

Cordova cli 9.0.0
cordova-android 8.0.0

plugins used are

cordova-plugin-app-version 0.1.9
cordova-cookie-emperor
cordova-plugin-customurlscheme 4.3.0 
cordova-plugin-device 2.0.3 
cordova-plugin-file  6.0.1
cordova-plugin-file-opener2
cordova-plugin-file-transfer 1.7.1
cordova-plugin-inappbrowser  3.0.0 
cordova-plugin-mediascanner  0.1.3 
cordova-plugin-splashscreen 5.0.2 
cordova-plugin-statusbar  2.4.2 
cordova-plugin-transport-security  0.1.2
cordova-plugin-whitelist 1.3.3 
cordova-plugin-zeep  0.0.4

## Checklist

- [x] I searched for existing GitHub issues
- [ ] I updated all Cordova tooling to most recent version
- [ ] I included all the necessary information above

Similar issue found https://github.com/apache/cordova/issues/95

**Comment by bandhavya** [comment]
This issue happens where there is config-file element REQUEST_INSTALL_PACKAGES in the config.xml

<config-file xmlns:android="http://schemas.android.com/apk/res/android" parent="/manifest" target="AndroidManifest.xml"> 
            <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES" /> 
        </config-file>

Without this config-file element all the permissions are properly added to AndroidManifest.xml

**Comment by Domvel** [comment]
I'm facing the same or a similar problem.
In this screenshot, you see two builds from the same sources. The `AndroidManifest.xml` from the APK (apktool).
![apk-manifest](https://user-images.githubusercontent.com/35452565/97688867-18027780-1a9b-11eb-86e3-efbdd2c57d62.png)

You can see that one build (`platforms\android\app\build\outputs\apk\debug\app-debug.apk`) has no `INTERNET` permission.
But the `BLUETOOTH_ADMIN` twice.

**Comment by Domvel** [comment]
I had the same problem. In this screenshot you see the `AdronidManifest.xml` of two builds. The permission `INTERNET` is sometimes overwritten by `BLUETOOTH_ADMIN`. In this case you see the bluetooth one twice. 

![apk-manifest](https://user-images.githubusercontent.com/35452565/97700959-f27e6980-1aac-11eb-88e6-c637fb251509.png)

**Solution**

Not sure if this is your problem, but I think there is a problem with "merge" in this case. See the `config.xml`:

This does not work:

```xml
<edit-config file="AndroidManifest.xml" mode="merge" target="/manifest/uses-permission" xmlns:android="http://schemas.android.com/apk/res/android">
  <uses-permission android:name="android.permission.INTERNET"/>
  <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
</edit-config>
```

This works:

```xml
<config-file target="AndroidManifest.xml" parent="/manifest" xmlns:android="http://schemas.android.com/apk/res/android">
  <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
  <uses-permission android:name="android.permission.INTERNET"/>
</config-file>
```

Not sure if this was added manually or automatically by Cordova. And not sure if this is the perfect solution.

**Comment by Domvel** [comment]
I had the same problem. In this screenshot you see the `AdronidManifest.xml` of two builds. The permission `INTERNET` is sometimes overwritten by `BLUETOOTH_ADMIN`. In this case you see the bluetooth one twice. 

![image](https://user-images.githubusercontent.com/35452565/97702910-0d061200-1ab0-11eb-8fc2-888a95a28d26.png)

**Solution**

Not sure if this is your problem, but I think there is a problem with "merge" in this case. See the `config.xml`:

This does not work:

```xml
<edit-config file="AndroidManifest.xml" mode="merge" target="/manifest/uses-permission" xmlns:android="http://schemas.android.com/apk/res/android">
  <uses-permission android:name="android.permission.INTERNET"/>
  <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
</edit-config>
```

This works:

```xml
<config-file target="AndroidManifest.xml" parent="/manifest" xmlns:android="http://schemas.android.com/apk/res/android">
  <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
  <uses-permission android:name="android.permission.INTERNET"/>
</config-file>
```

Not sure if this was added manually or automatically by Cordova. (Have to check this by creating a new project.) And not sure if this is the perfect solution.

**Comment by mbaileys** [comment]
This is an ongoing issue. It took us a while to figure out INTERNET permission had to be added manually in cordova-android 9.

**Comment by Joannanas** [comment]
I just struggled for a few days with a related and pretty annoying issue, and my conclusion might help to isolate the case. Everytime I run the "cordova run android" command line, it detects some non-existent conflict for the AndroidManifest and deletes one by one every permission added by the phonegap-barcodescanner used by my app, along with one of the "edit-config" from the cordova-plugin-push (this is what made me question everything)

I thought that maybe I add something written in my config.xml file that really made a conflict, but I have no permission change in it, so the content of the config.xml alone is not the issue here.

I tried the verbose option but it does not explicitly tell what is in conflict, so i tried to comment everything I have in my config.xml file, and I figured out that the conflict comes from trying to merge config into the application as follow : 
```xml 
<edit-config file="AndroidManifest.xml" mode="merge" target="/manifest/application">
            <application android:theme="@style/AppTheme"/>
</edit-config>
```

It seems also relevant to say that I do not use the option "--force" when adding a plugin, but I still only have this small warning line "Conflict found, edit-config changes from config.xml will overwrite plugin.xml changes" everytime I use the "cordova run android" command, ONLY when I have the previous merge in my config.xml file. If I comment it, no conflict is detected and all my permissions/edit-config stay in the generated AndroidManifest.

So, TLDR, my permissions are deleted when doing `cordova run android`command by the fact that I have an edit-config trying to merge an atrribute to my application tag. The edit-config is then done, but every run command deletes some line from the generated AndroidManifext.xml.
